app/core/RPCMethods/createUnsupportedMethodMiddleware/index.ts

```diff
@@ -0,0 +1,20 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { JsonRpcMiddleware } from 'json-rpc-engine';
+import { UNSUPPORTED_RPC_METHODS } from '../utils';
+
+/**
+ * Creates a middleware that rejects explicitly unsupported RPC methods with the
+ * appropriate error.
+ */
+const createUnsupportedMethodMiddleware = (): JsonRpcMiddleware<
+  unknown,
+  void
+> =>
+  async function unsupportedMethodMiddleware(req, _res, next, end) {
+    if ((UNSUPPORTED_RPC_METHODS as Set<string>).has(req.method)) {
```

Comment:
Would this make it more type safety? For me it's okay to go like this, just wondering 
It also increases the burden of adding new unsupported rpc methods in two places (The set constant and the type here)

```suggestion
    if ((UNSUPPORTED_RPC_METHODS as Set<'eth_signTransaction'>).has(req.method)) {
```

Reply:
The problem is that `UNSUPPORTED_RPC_METHODS` is of type `Set<'eth_signTransaction'>` while `req.method` is `string`. The code is good as-is.